### DISCUSSION TITLE
Configuring IDE for intellisense in Gizmo code

### DISCUSSION THREAD

**Discussion by Ewall198** [godot-jolt/godot-jolt#971] [Questions] [answered]
While hacking and contributing, how do you configure the IDE (VS'22 or VSCode) so that you have proper intellisense for the code in the Gizmo.

**Comment by Ewall198** [comment]
I was using VSCode. After doing a bit more research, it'll probably be easier Visual Studio. Going to give it another shot tomorrow.

**Comment by mihe** [comment]
I mostly use VS Code myself these days. You should just need to install these two extensions (and Visual Studio 2022 for the toolchain):

- [ms-vscode.cpptools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
- [ms-vscode.cmake-tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools)

... and you should get the option in VS Code to pick from the configure/build presets specified in `CMakePresets.json`.

EDIT: Oh, and you likely need to add this to your VS Code's `settings.json`: `"C_Cpp.default.configurationProvider": "ms-vscode.cmake-tools"`

**Reply by Ewall198** [reply]
I have both of those extensions and I added `"C_Cpp.default.configurationProvider": "ms-vscode.cmake-tools"` to settings.

When I select "active configure preset", I use `[MSVC, x64]`
When I go to select "active build preset", the only option I see is `[Default]` and `Add a New Preset...`. Clicking `Add a New Preset...` allows me to add build presets to `CMakePresets.json`, which isn't what I'm trying to do.
But, I'm still not sure how to select the build preset used for intellisense.

**Reply by Ewall198** [reply]
For clarity, in the CMake sidebar under `Configure` the options are all of the format `<compiler>-<architecture>`
Under `Build` when trying to change the active preset from `[Default]`, the only option is `[Default]` or `Add a New Preset...`.

**Reply by mihe** [reply]
> When I go to select "active build preset", the only option I see is `[Default]`

Hm, yeah, that shouldn't be like that. That's weird. It looks like [this part](https://github.com/godot-jolt/godot-jolt/blob/202b428527815031b5c32e9ba2d065645ec70523/CMakePresets.json#L372-L376) is what's breaking things.

If I just remove those `${hostSystemName}` checks from all the base build configurations then everything just works as expected, which is a bit strange. I could have sworn those were necessary at some point. Now instead it seems to infer the `${hostSystemName}` constraint from the configure preset. It's been a while since I wrote that stuff though, so I guess things could have changed since then.

The reason I hadn't seen this myself is because I use a `CMakeUserPresets.json` (see [here](https://github.com/godot-jolt/godot-jolt/blob/master/docs/hacking.md#user-presets)) and pretty much only ever use the `dev-*` presets found there, which don't have any `${hostSystemName}` checks in them.

I'll push a fix for it shortly.

**Reply by mihe** [reply, marked as answer]
The removal of the `${hostSystemName}` conditions is merged now.

**Reply by Ewall198** [reply]
Yup, that fixed it. I just had to pull in the changes then select the Config and Build and then I had to build it once in VSCode and now it works properly. Thanks!
